test/jdk/java/awt/PopupMenu/PopupMenuLocation.java

```diff
@@ -119,9 +126,22 @@ private static void openPopup(final Frame frame) throws Exception {
         robot.mousePress(InputEvent.BUTTON1_DOWN_MASK);
         robot.mouseRelease(InputEvent.BUTTON1_DOWN_MASK);
         robot.waitForIdle();
-        if (!action) {
-            throw new RuntimeException();
+        if (!actionEventReceivedLatch.await(5, TimeUnit.SECONDS)) {
+            captureScreen();
```

Comment:
So if it hasn't passed in 5 seconds, you fail the test. How is that improving stability ?

Reply:
ping ?